components/Authors.tsx

```diff
@@ -0,0 +1,289 @@
+"use client";
+import { useMutation, useQuery } from "@apollo/client";
+import { useState } from "react";
+import AuthorModal from "./AuthorModal";
+import BookModal from "./BookModal";
+import { AUTHORS_QUERY, DELETE_AUTHOR } from "@/lib/gql/author";
+import Modal from "./Modal";
+import Pagination from "./Pagination";
+import { Author } from "@/lib/types/author";
+import { useSearchParams, useRouter } from "next/navigation";
+import GenericErrorModal from "./GenericErrorModal";
+
+const ITEMS_PER_PAGE = 10;
+
+const ConfirmDeleteModal = ({
```

Comment:
1. `ConfirmDeleteModal` should be in it's own file.
2. Should be reused in both `Authors.tsx` and `Books.tsx`